### ISSUE TITLE
List view: the 'sticky' information is only visual

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#59409]
### Description

In the List view, some icons may be shown to provide information about some block properties, for example:
- when the block is locked.
- when the block is sticky

Screenshot:

![Screenshot 2024-02-26 at 14 34 26](https://github.com/WordPress/gutenberg/assets/1682452/fe2b7f34-5e45-49ec-a115-8f1bdd5f938d)

While for the 'locked' state a text `(locked)` is appended to the block item aria-label, there's no textual information for the 'sticky' case.

As such, there's no parity between the visual information provided to sighted users and the textual information provided to users with no vision or low vision or users who don't recognize icons meaning.

Ideally, this kind of important information should be provided by the means of _visible_ text.

At the very least, the 'sticky' state should be communicated by the means of some visually hidden text or description

### Step-by-step reproduction instructions

- Go to teh post editor.
- Add a group block with a paragraph.
- First, make the group block sticky.
- Then, make the group block locked.
- Open the list view.
- Observe the group block item has two icons: sticky and lock.
- Inspect the source.
- Oserve the aria-label of the group block is: `Group (locked)`.
- Onserv there's no textual information of any type about the sticky state.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by t-hamano** [comment]
I think that in addition to the sticky information, the anchor text also needs to be conveyed to the screen reader. I'm at a loss as to what is the appropriate way to convey this when these things are combined...

![image](https://github.com/WordPress/gutenberg/assets/54422211/58311709-3781-4f78-9220-b7e88fcf1966)

**Comment by afercia** [comment]
Thanks @t-hamano. Yes I wanted to investigate other potential icon / things that may be rendered within the list view items. The `anchor` is a good one. I also want to investigate the image previews.
Any more potential things that may be shown there you can think of?

**Comment by t-hamano** [comment]
Looking at [the List View implementation](https://github.com/WordPress/gutenberg/blob/0b5cd48bfb474377e6947a7c3302d37bd910a586/packages/block-editor/src/components/list-view/block-select-button.js#L271-L307), there are four pieces of information that may be displayed in addition to the block name:

- Anchor text
- Sticky icon
- Image thumbnails
- Locked icon

Currently, only theGroup block supports sticky position, but image thumbnails will not be displayed in this block. However, assuming that image thumbnails are also supported in the future, it is possible that all four pieces of information will be displayed.

In that case, will it be a long label like `Group (Locked, sticked, anchor text: test, has 4 images)`?

**Comment by afercia** [comment]
Regarding the 'anchor' information, I wonder how displaying truncated text that doesn't actually help understand what the anchor is, is any useful for users. Also, there's potentially really too much stuff and info provided within the list view items. 

Overall, I think hiding or truncating text 'just to make it fit into the design` is less than ideal. Content should never be hidden. The interface should be _designed around the content_ and not vice-versa.

Each list view item can get to a state where the UI is really not useful. Example screenshot:

![Screenshot 2024-03-05 at 16 48 33](https://github.com/WordPress/gutenberg/assets/1682452/1326c795-164d-4259-968a-dd38b9bb1882)

**Comment by t-hamano** [comment]
I agree that depending on the block settings, the current list view displays too much information. We may need to reconsider the information that should be displayed in the list view and its ideal state.

cc @gutenbergplugin

**Comment by afercia** [comment]
With the new Bindings API, there's now a new information provided by only icon: Bound blocks will display this purple icon:

![Screenshot 2024-03-11 at 12 37 52](https://github.com/WordPress/gutenberg/assets/1682452/9b7d1a20-f5d6-483f-bb21-ff4cbebb6a07)

There is no textual information in this case as well.

From a techinical perspective, the current code implementation appears to be less than ideal:

https://github.com/WordPress/gutenberg/blob/f7070e8a50b7b544588fc98caa53fe5109a431ca/packages/block-editor/src/components/list-view/block-select-button.js#L290-L321

- All icons have no textual alternative information.
- The Sticky icon is the only icon that is supposed to have a Tooltip. Not sure why the `Tooltip` component is used only for this icon.
- Tooltips don't work with icons anyways, so it appears this hasn't been tested sufficiently. Edit: see better explanation in the following comment.
- The image preview is a background image annd it's also ARIA hidden.

**Comment by afercia** [comment]
> Tooltips don't work with icons anyways,

Turns out the tooltip on the 'sticky position' icon stopped working after https://github.com/WordPress/gutenberg/pull/57202 so this is a regression from WordPress 6.4 where this tooltip does work. Cc @ciampo not sure this is the intended behavior of the check for nested tooltips.

Screenshot:

![Screenshot 2024-03-11 at 14 47 35](https://github.com/WordPress/gutenberg/assets/1682452/76c7d48c-b845-4db2-b6d0-8363fd95387d)

We should remove the tooltip anyways because:
- It is actually set on an Icon _inside a link_. Making the tooltip work again would produce a focusable element inside a link ie.: two nested focusable elements, which is far from ideal. This can be checked in WordPress 6.4 where the tooltip works and Ariakit adds a `tabindex="0"` to the SVG icon.
- As mentioned in other issues, the fact Ariakit adds a `tabindex="0"` to any element where a Tooltip is set to is far from ideal because it will make unsemantic elements with tooltips focusable e.g. focusable `div` elements and such.
- Tooltips should only be used to visually expose an interactive control _name_. They should never be used for descriptions.
- There is no textual information anyways so the Tooltip is pointless.

**Comment by ciampo** [comment]
Sorry for the delay, still catching up after having been AFK for a while.

@afercia as you point out, `Tooltip` should not be rendered inside an interactive element (an anchor tag). This is an incorrect use of the component, which leads to inconsistent and unexpected results. I've opened #63850 to remove it.

**Comment by t-hamano** [comment]
Are there any other issues that need to be addressed regarding this issue?

- Block Bindings icon has been removed from the list view: [#59477](https://github.com/WordPress/gutenberg/pull/59477)
- Do we need to provide any information to images?

**Comment by ciampo** [comment]
@afercia may be AFK right now, but looking through his feedback, I think that a good solution could be to make sure:

- that all icons are presentational only;
- that the meaning of every icon is reflected in the list item's description text (like I did for the `position` icon in https://github.com/WordPress/gutenberg/pull/63850)

Do we already have descriptions for all possible icons that can be displayed in the list view?

**Comment by t-hamano** [comment]
Thanks for the reply.

I re-read this issue and it seems that the information other than the block name displayed in the list item, and whether it is read out properly, is as follows:

- ❌ Anchor text
- ✅ Sticky icon
- ❌ Image thumbnails
- ✅ Locked icon

So it might be best to leave this issue open for now.
